payjoin-test-utils/src/lib.rs

```diff
@@ -236,15 +230,17 @@ const TESTS_TIMEOUT: Duration = Duration::from_secs(20);
 const WAIT_SERVICE_INTERVAL: Duration = Duration::from_secs(3);
 
 pub async fn wait_for_service_ready(
-    service_url: Url,
+    service_url: &str,
     agent: Arc<Client>,
 ) -> Result<(), &'static str> {
-    let health_url = service_url.join("/health").map_err(|_| "Invalid URL")?;
+    let health_url: Url = format!("{}/health", service_url.trim_end_matches("/"))
+        .parse()
+        .map_err(|_| "Invalid Url")?;
```

Comment:
```suggestion
    let health_url = format!("{}/health", service_url.trim_end_matches("/"));
```

Because the `fn get` type signature takes `U: IntoUrl` and `From<String> for [reqwest::]IntoUrl` is implemented.

```rs
reqwest::async_impl::client::Client
pub fn get<U>(&self, url: U) -> RequestBuilder
where
    U: IntoUrl,
```